internal/webhooks/validate.go

```diff
@@ -0,0 +1,64 @@
+package webhooks
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	addonsv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+)
+
+func validateInstallSpec(addon addonsv1alpha1.Addon) error {
```

Comment:
Would you mind adding a unit test for this function?
The integration tests are awesome, but it's way easier and faster to test all validation cases via a unittests.